### PR TITLE
DEV-288 Fix for integer commit sha bug

### PR DESCRIPTION
**Pull request by robottaway**

### CODE REVIEW

File: lib/pd-cap-recipes/tasks/git.rb [hunk 1]
```diff
@@ -206,13 +210,24 @@ def git_sanity_check(tag)
       return
     end
 
+    if not is_already_deployed
+      Capistrano::CLI.ui.say yellow "It appears you have not deployed yet, skipping sanity check"
```
Comment (margueritepd) [inline lib/pd-cap-recipes/tasks/git.rb]: "never deployed this app"?

**Comment by margueritepd** [comment]
Seems reasonable :cherries:
aside from the attention demanded from :gem: :cop: (our lint robot overlord)

File: lib/pd-cap-recipes/tasks/git.rb [hunk 2]
```diff
@@ -253,7 +276,13 @@ def safe_current_revision
...
       logger.info "*" * 80
-      nil
+      return nil
     end
   end
+
+  # If a current revision exists we assume we've deployed before.
+  def is_already_deployed
+    return false if safe_current_revision.nil?
+    return true
```
Comment (dougbarth) [inline lib/pd-cap-recipes/tasks/git.rb]: Pet peeve: boolean math. This method can just be written like this.

```ruby
def is_already_deployed
  return ! safe_current_revision.nil?
end

# or

def is_already_deployed
  !! safe_current_revision
end
```

Comment (alperkokmen) [inline lib/pd-cap-recipes/tasks/git.rb]: even better without the space after `!!` :smile:

**Comment by margueritepd** [comment]
Looks good! Thinking we could improve the readability of some warnings:

- "It appears you have not deployed yet, skipping sanity check" -> "It appears you have never deployed this app yet, continuing without sanity checking revision to deploy" (if I'm understanding `current_revision` correctly)
- "Skipping reverse deploy check flag found and non-production environment." -> "reverse_deploy_ok - continuing without sanity checking revision to deploy"

File: lib/pd-cap-recipes/tasks/git.rb (hunk 2 above)
Comment (robottaway) [inline lib/pd-cap-recipes/tasks/git.rb]: Thanks all I'll make this change and have it pushed soon.

**Pull request merged**